src/main/java/software/amazon/cloudwatchlogs/emf/model/MetricDefinition.java

```diff
@@ -43,18 +47,36 @@ class MetricDefinition {
     @JsonDeserialize(using = UnitDeserializer.class)
     private Unit unit;
 
+    @Getter
+    @Setter
+    @JsonProperty("StorageResolution")
+    @JsonInclude(
+            value = JsonInclude.Include.CUSTOM,
+            valueFilter =
+                    StorageResolutionFilter.class) // Do not serialize when valueFilter is true
+    @JsonSerialize(using = StorageResolutionSerializer.class)
+    public StorageResolution storageResolution;
+
     @JsonIgnore @NonNull @Getter private List<Double> values;
 
     MetricDefinition(String name) {
-        this(name, Unit.NONE, new ArrayList<>());
+        this(name, Unit.NONE, StorageResolution.STANDARD, new ArrayList<>());
     }
 
     MetricDefinition(String name, double value) {
-        this(name, Unit.NONE, value);
+        this(name, Unit.NONE, StorageResolution.STANDARD, value);
     }
 
     MetricDefinition(String name, Unit unit, double value) {
-        this(name, unit, new ArrayList<>(Arrays.asList(value)));
+        this(name, unit, StorageResolution.STANDARD, new ArrayList<>(Arrays.asList(value)));
+    }
+
+    MetricDefinition(String name, StorageResolution storageResolution, double value) {
+        this(name, Unit.NONE, storageResolution, new ArrayList<>(Arrays.asList(value)));
+    }
+
+    MetricDefinition(String name, Unit unit, StorageResolution storageResolution, double value) {
```

Comment:
Can we change the order of this parameter such as name, unit, value, storageresolution.

Reply:
I wrote the definition like this to maintain consistency in the class for this method. 

If you look at existing method signatures, 'value' parameter is always at the end. These three are existing methods:
1. MetricDefinition(String name) 
2. MetricDefinition(String name, double value)
3. MetricDefinition(String name, Unit unit, double value) ([Link](https://github.com/awslabs/aws-embedded-metrics-java/pull/129/files/f4348977813388b799b068f5df0bd62c3c15163d#diff-e09a6172c54928abe954a81dcefe2ebb617251b90c6ce0dcc9267113d3123eaeR70))

Hence, I added these two:
1. MetricDefinition(String name, StorageResolution storageResolution, double value)
2. MetricDefinition(String name, Unit unit, StorageResolution storageResolution, double value)

Reply:
As agreed, this can be resolved.